cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -4848,6 +4848,9 @@ macro(build_awssdk)
     # Link time optimization is causing trouble like #34349
     string(REPLACE "-flto=auto" "" AWS_LC_C_FLAGS "${AWS_LC_C_FLAGS}")
     string(REPLACE "-ffat-lto-objects" "" AWS_LC_C_FLAGS "${AWS_LC_C_FLAGS}")
+    # Added by R with --enable/use-lto see #39523
+    string(REPLACE "-flto" "" AWS_LC_C_FLAGS "${AWS_LC_C_FLAGS}")
+    string(REPLACE "-flto=10" "" AWS_LC_C_FLAGS "${AWS_LC_C_FLAGS}")
```

Can we remove all `-flto*` flags?

```suggestion
    # "-flto" and "-flto=10" are added by R with --enable/use-lto see #39523
    string(REPLACE "-flto[^ ]*" "" AWS_LC_C_FLAGS "${AWS_LC_C_FLAGS}")
    string(REPLACE "-ffat-lto-objects" "" AWS_LC_C_FLAGS "${AWS_LC_C_FLAGS}")
```

---

Sounds like a future proof idea :)